### PR TITLE
Changes from Realm Roles UX Review [List]

### PR DESCRIPTION
**Pull request by jenny-s51**
## Motivation

#416 

## Brief Description

Towards #416 - Addresses all items in [List] category of UX Review

- Removes the margin of the Role list, reference the latest style of the Client list.
- Removes the margin of the Associated Role list
- Search box in the toolbar: search now works by hitting enter
- When searching for something that doesn't have a matched result, show the search empty state (https://github.com/keycloak/keycloak-admin-ui/pull/435)
- Adds chip group functionality to Toolbar+DataTable for filtering by category in other sections of Keycloak
- Fixes save button functionality on Role Details page

## Verification Steps

Verify that all the fixes listed above have been addressed 

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [x] Help has been implemented
- [ ] axe report has been run and resulting a11y issues have been resolved
- [x] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

**Review by edewit — COMMENTED** [review]
I'm confused about these `filterChips` how are they supposed to work? I thought it would be that when I search for one term and then search for something else after that these terms would end up beneath the search input?

**Comment by yih-wang** [comment]
> I'm confused about these `filterChips` how are they supposed to work? I thought it would be that when I search for one term and then search for something else after that these terms would end up beneath the search input?

Each time the user clicks the search, the term will show as a chip beneath the toolbar. Reference this demo - https://www.patternfly.org/v4/components/table/react-demos/#filterable

**Review by mfrances17 — CHANGES_REQUESTED** [review]
Aside from the filter chips, which we are handling as a separate issue, only one of the issues from the bulleted list of fixed items wasn't working in my testing. I am still getting the empty state when searching for a string that does not exist:

search

result

**Comment by jenny-s51** [comment]
@mfrances17 thank you for your comment - I rebased and search empty state should now be addressed (https://github.com/keycloak/keycloak-admin-ui/pull/435). Please kindly re-review when you have a moment :)

File: src/components/table-toolbar/KeycloakDataTable.tsx [hunk 1]
```diff
@@ -170,32 +170,7 @@ export function KeycloakDataTable<T>({
...
-    setFilteredData(
-      rows!.filter((row) =>
-        row.cells.some(
-          (cell) =>
-            cell &&
-            getNodeText(cell).toLowerCase().includes(search.toLowerCase())
-        )
-      )
-    );
-  };
```
Comment (edewit) [inline src/components/table-toolbar/KeycloakDataTable.tsx:198]: don't we still need this?

**Review by edewit — CHANGES_REQUESTED** [review]
Filtering still needed right?

File: src/components/table-toolbar/KeycloakDataTable.tsx (hunk 1 above)
Comment (jenny-s51) [reply src/components/table-toolbar/KeycloakDataTable.tsx:198]: `filter` (which calles 'getNodeText`) is never used so eslint throws an error... this function was used to filter in real-time as the user types, right?

Per @yih-wang:

> It's better to keep consistency among all the search behaviors in a console. I looked into some other websites and think it's a common usage that return to the list by deleting the search input and search again. So let's keep using that. I update the clarification of the behaviors in this issue https://github.com/keycloak/keycloak-admin-ui/issues/417

**Comment by jenny-s51** [comment]
@edewit filter (which calls `getNodeText`) is never used here so eslint throws an error... this function was used to filter in real-time as the user types

Per @yih-wang:

> It's better to keep consistency among all the search behaviors in a console. I looked into some other websites and think it's a common usage that return to the list by deleting the search input and search again. So let's keep using that. I update the clarification of the behaviors in this issue #417

**Review by mfrances17 — APPROVED** [review]
The issue I had flagged about empty state on not finding a role name search result is now fixed... LGTM. Nice job.

File: src/components/table-toolbar/KeycloakDataTable.tsx (hunk 1 above)
Comment (edewit) [reply src/components/table-toolbar/KeycloakDataTable.tsx:198]: Yes it was used to filter in real time, but we only want to change it that we no longer filter in real time not that we don't filter at all. If you remove this searching on the client scope page will no longer work.

**Review by edewit — CHANGES_REQUESTED** [review]
we still need filtering just not the while you type.

**Comment by jenny-s51** [comment]
@edewit that makes sense - thank you for clarifying! Updated to use the filter function again

**Review by edewit — APPROVED** [review]

**Pull request merged**